### ISSUE TITLE
OAuth documentation

### ISSUE THREAD

**Issue by rwentzel9** [rickbergfalk/sqlpad#160]
I setup sqlpad according to the instructions provided, and everything seems to work great except for the OAuth integration. I was careful to make sure everything is setup as you instructed, but regardless I'm stuck at a code 500 Internal Server Error when Google redirects me back with the code value attached to the URL.

Here is the error I'm seeing in the console when this happens:

InternalOAuthError: failed to fetch user profile
    at C:\Users\[redacted]\AppData\Roaming\npm\node_modules\sqlpad\node_modules\pa
ssport-google-oauth2\lib\oauth2.js:92:28
    at passBackControl (C:\Users\[redacted]\AppData\Roaming\npm\node_modules\sqlpa
d\node_modules\oauth\lib\oauth2.js:132:9)
    at IncomingMessage.<anonymous> (C:\Users\[redacted]\AppData\Roaming\npm\node_m
odules\sqlpad\node_modules\oauth\lib\oauth2.js:157:7)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:974:12)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)

Any idea about what might be the issue here? I'm attempting to test it on localhost and have tried several configurations, but nothing seems to work. Any help would be appreciated, or if there is any additional documentation on the OAuth feature, that'd be incredible; it's hardly mentioned in the current documentation, and I'm sure I'm not the only who has been confused about its functionality.

Thanks!

**Comment by rickbergfalk** [comment]
Hi @rwentzel9 - yeah the oauth documentation is lacking. Sorry about that. When I tested locally here's what I had used in my Google API oauth client credentials config:

- **Authorized JavaScript origins:** http://localhost:8080
- **Authorized redirect URIs:** http://localhost:8080/auth/google/callback

Also be sure to **enable the Google+ API** for your Google API project. If this isn't enabled it might be why the user profile isn't being fetched. 

Some other config settings to consider for things to be fully useful:
PUBLIC_URL=http://localhost     (I don't think the port is necessary here when testing locally)
DISABLE_USERPASS_AUTH=true  (disables local user logins)
WHITELISTED_DOMAINS=yourgoogledomain.com      (if you don't want to have to whitelist users)

Hope this helps - let me know how it goes
